### DISCUSSION TITLE
Can someone please enlighten me on the compiler install location

### DISCUSSION THREAD

**Discussion by netskink** [zopencommunity/meta#164] [Q&A]
I am working with the system programming team to have our compiler installed in the proper place for a USS setup.

In USS, do you have:

```
/usr/lpp/IBM/cnw/v1r1/openxl/bin/ibm-clang
/usr/lpp/IBM/cnw/v1r1/openxl/bin/ibm-clang++
/usr/lpp/IBM/cnw/v1r1/openxl/bin/ibm-clang++64
/usr/lpp/IBM/cnw/v1r1/openxl/bin/ibm-clang-cpp
/usr/lpp/IBM/cnw/v1r1/openxl/bin/ibm-clang64
```

And do you have the includes for the open compiler in 

```
/usr/lpp/IBM/cnw/v1r1/openxl/include/c++/v1
```
Such that the stdlib.h is `/usr/lpp/IBM/cnw/v1r1/openxl/include/c++/v1/stdlib.h`?

Or, does it reside as `/usr/include/c++/v1/stdlib.h`

**Comment by MikeFultonDev** [comment]
Hm. I just saw this - apologies @netskink . Have you sorted this out now?

**Reply by netskink** [reply]
Yes I put this in the attachment for my email.  It wouldn't hurt though to detail what compilers were used for those wanting to build from source though.  I realize the compiler build checks use the _compiler_version define or however it's spelled but it's not obvious until you get into the source.

**Reply by MikeFultonDev** [reply]
Ok. Good point. Except for a couple cases where I had to build with clang I think we are using zlclang everywhere. Having said that, @IgorTodorovskiIBM - I know you are doing work here so I will let you comment

**Reply by IgorTodorovskiIBM** [reply]
We're currently building with xlclang, but will be moving to clang and in particular this version of clang: https://epwt-www.mybluemix.net/software/support/trial/cst/programwebsite.wss?siteId=1803
We have this issue: https://github.com/ZOSOpenTools/meta/issues/226 to track the migration.

This version of clang installs to /usr/lpp/IBM/oelcpp/v2r0. It also has a ./include/c++/v1/stdlib.h header file.

What tool are you trying to build in particular?

**Reply by netskink** [reply]
This original comment was regarding the open tools compiler.  I was trying to build v8 as a main goal, but perhaps it was encountered when building zoslib as a prereq for v8.

This has all be OBE though.  I've learned to use the `zopen install <package>` method for the prereq tools to build v8.  From the summary attachment in my email, here is some edited excerpts for the workflow:

### Install z/OS Open Tools


Use this [workflow](https://zosopentools.github.io/meta/#/) to install z/OS Open Tools. Basically, use `zopen-setup` script to setup `~/zopen/boot` and then use `zopen install <package>` to install additional packages in the toolsuite.  For example:

```
$ cd ~/zopen/dev
$ zopen isntall zoslib
***WARNING: Run zopen init to configure the install location. Using current working directory...
***WARNING: Setting ZOPEN_GIT_OAUTH_TOKEN is recommended to ensure that you do not hit the GitHub API cap. See --help for more details.
Installing zoslib...
New release with tag "zoslibport_693" found for zoslibport
Preparing to download zoslib
Downloading latest release from zoslibport...
Extracting zoslib-zopen.20230306_191219.zos.pax.Z...
Setting up zoslib...
Setup completed.
Successfully installed zoslib to /z/xxxxx/zopen/dev/zoslib/
```

In `~.bashrc`, do this:

```
#
# zopen bootenv
#
pushd .
cd ${HOME}/zopen/boot
##source ${HOME}/zopen/boot/.bootenv
source .bootenv
popd

# add path for z/OS open source components I've installed after the fact
export PATH=/z/xxxxx/zopen/prod/man/bin:$PATH
export PATH=/z/xxxx/zopen/dev/ninja/bin:$PATH
export PATH=/z/xxxx/zopen/dev/depot_tools:$PATH
export PATH=/z/xxxxx/zopen/dev/patch/bin:$PATH
```

The tools installed via `zopen install <package>` in the `dev` folder need the
stanza for the path mod above.  The other packages appear to updated in the PATH
when the .bootenv script it invoked.  (TODO:Hmm, I should look and see if there is a similar
.devenv script in the .../zopen/dev folder when a package is installed in .../zopen/dev.)

**Reply by IgorTodorovskiIBM** [reply]
Sounds good, one thing to note is that the zoslib port is based off of the _zopen_ branch which has a few additions compared to the main branch which was used to build v8.
